libs/events/import/api/src/lib/import/agenda-basel.service.spec.ts

```diff
@@ -0,0 +1,148 @@
+import {Cache} from 'cache-manager'
+import {HttpService} from '@nestjs/axios'
+import {ImageFetcherService, MediaAdapterService} from '@wepublish/image/api'
+
+describe('AgendaBaselService', () => {
+  let service: AgendaBaselService
+  let prismaClient: PrismaClient
+  let cacheManager: Cache
+  let httpClient: HttpService
+
+  beforeAll(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        AgendaBaselService,
+        {
+          provide: EVENT_IMPORT_PROVIDER,
+          useValue: [AgendaBaselService]
+        },
+        {
+          provide: HttpService,
+          useValue: {
+            get: jest.fn()
+          }
+        },
+        {
+          provide: CACHE_MANAGER,
+          useValue: {
+            get: jest.fn(),
+            set: jest.fn()
+          }
+        },
+        {
+          provide: PrismaClient,
+          useValue: {
+            event: {
+              create: jest.fn()
+            },
+            image: {
+              create: jest.fn()
+            }
+          }
+        },
+        {
+          provide: ImageFetcherService,
+          useValue: {
+            fetch: jest.fn()
+          }
+        },
+        {
+          provide: MediaAdapterService,
+          useValue: {
+            uploadImageFromArrayBuffer: jest.fn()
+          }
+        }
+      ]
+    }).compile()
+
+    service = module.get<AgendaBaselService>(AgendaBaselService)
+    cacheManager = module.get<Cache>(CACHE_MANAGER)
+    prismaClient = module.get<PrismaClient>(PrismaClient)
+    httpClient = module.get<HttpService>(HttpService)
+  })
+
+  const mockEvent: Event = {
+    id: '1',
+    createdAt: new Date(),
+    modifiedAt: new Date(),
+    name: 'Event 1',
+    description: [],
+    status: EventStatus.Scheduled,
+    location: '',
+    externalSourceId: '',
+    externalSourceName: '',
+    startsAt: new Date(),
+    endsAt: undefined
+  }
+
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  describe('AgendaBaselService', () => {
+    test('importedEvents method should return imported events from the cache if available', async () => {
+      const mockEventsDocument = {
+        nodes: [mockEvent],
+        pageInfo: {endCursor: '1', hasNextPage: false, hasPreviousPage: false, startCursor: '1'},
+        totalCount: 1
+      }
+      jest.spyOn(cacheManager, 'get').mockResolvedValueOnce([mockEvent])
+
+      jest.spyOn(service, 'importedEvents').mockResolvedValueOnce(mockEventsDocument)
+      const result = await service.importedEvents({
+        filter: {} as any,
+        order: 1,
+        skip: 0,
+        take: 10,
+        sort: ImportedEventSort.CREATED_AT
+      })
+      expect(result).toEqual(mockEventsDocument)
+    })
+
+    test('importedEvent should return the imported event by ID from the corresponding provider', async () => {
+      const id = '1'
+
+      jest.spyOn(service, 'importedEvent').mockResolvedValueOnce(mockEvent)
```

Comment:
This test doesn't do anything as the thing that is being tried to test, is mocked